nodewatcher/modules/devices/ubnt/edgerouter.py

```diff
@@ -0,0 +1,54 @@
+from django.utils.translation import ugettext_lazy as _
+
+from nodewatcher.core.generator.cgm import base as cgm_base, protocols as cgm_protocols, devices as cgm_devices
+
+
+class UBNTEdgerouterX(cgm_devices.DeviceBase):
+    """
+    UBNT EdgerouterX device descriptor.
+    """
+
+    identifier = 'ub-edgerouter-x'
+    name = "Edgerouter X"
+    manufacturer = "Ubiquiti"
+    url = 'http://www.ubnt.com/'
+    architecture = 'ramips_mt7621'
+    switches = [
+        cgm_devices.Switch(
+            'sw0', "Switch0",
+            ports=5,
+            cpu_port=6,
+            vlans=16,
+            configurable=True,
+            presets=[
+                cgm_devices.SwitchPreset('default', _("Default VLAN configuration"), vlans=[
+                    cgm_devices.SwitchVLANPreset(
+                        'lan0', "Lan0",
+                        vlan=1,
+                        ports=[0, 1, 2, 3, 4],
+                    )
+                ])
+            ]
+        )
+    ]
+    ports = [
+        cgm_devices.EthernetPort('wan0', "Wan0"),
+    ]
+    port_map = {
+        'openwrt': {
+            'sw0': cgm_devices.SwitchPortMap('switch0', vlans='eth0'),
```

Comment:
Well that be beyond my understanding but I will try to figure it out.
Here is switch configuration in Lede source
https://github.com/lede-project/source/blob/master/target/linux/ramips/base-files/etc/board.d/02_network

ucidef_add_switch "switch0" \
			"1:lan" "2:lan" "3:lan" "4:lan" "0:wan" "6@eth0"